sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOReadImplementationCompatibilityTest.java

```diff
@@ -0,0 +1,145 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.beam.sdk.io.kafka.KafkaIOReadImplementationCompatibility.KafkaIOReadProperties;
+import org.apache.beam.sdk.io.kafka.KafkaIOTest.ValueAsTimestampFn;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.CaseFormat;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Sets;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class KafkaIOReadImplementationCompatibilityTest {
+
+  @Rule public final transient TestPipeline p = TestPipeline.create();
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testKafkaIOReadPropertiesEnumValuePresence() {
+    final Set<String> properties = getGetterPropertyNamesInUpperUnderscore(KafkaIO.Read.class);
+    final Set<String> enums = getEnumValueNamesInUpperUnderscore(KafkaIOReadProperties.class);
+
+    final Sets.SetView<String> missingEnums = Sets.difference(properties, enums);
+    final Sets.SetView<String> unnecessaryEnums = Sets.difference(enums, properties);
+
+    assertThat("There are missing 'KafkaIOReadProperties' enum values!", missingEnums, is(empty()));
+    assertThat(
+        "There are unnecessary 'KafkaIOReadProperties' enum values present!",
+        unnecessaryEnums,
+        is(empty()));
+  }
+
+  private static Set<String> getGetterPropertyNamesInUpperUnderscore(Class<?> clazz) {
+    final Set<String> result = Sets.newLinkedHashSet();
+    for (Method method : clazz.getDeclaredMethods()) {
+      if (!Modifier.isStatic(method.getModifiers())) {
+        if (!Void.TYPE.equals(method.getReturnType())) {
+          if (method.getParameterCount() == 0) {
+            for (String prefix : new String[] {"get", "is"}) {
+              final String methodName = method.getName();
+              if (methodName.startsWith(prefix) && methodName.length() > prefix.length()) {
+                final String propertyName = methodName.substring(prefix.length());
+                result.add(CaseFormat.UPPER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, propertyName));
+              }
+            }
+          }
+        }
+      }
+    }
+    return result;
+  }
+
+  private static Set<String> getEnumValueNamesInUpperUnderscore(
+      Class<? extends Enum<?>> enumClazz) {
+    return Stream.of(enumClazz.getEnumConstants()).map(Enum::name).collect(Collectors.toSet());
```

Comment:
errorprone and checkstyle both seem to be complaining about this line with: 
```
found raw type: Enum.
```

https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/22103/error-prone/new/source.74444392-91e4-4a9a-9a2f-1d3b119e3c22/#89

(There also seem to some red herrings in SnowflakeIO, I'm not sure what that's about)

Reply:
I gotta say these tests are rather flaky. (Just check which ones failed during the last 3 commit.) Anyway it all seems to pass now.